Size "mediumint" wasn't enough

Quix0r opened the pull request:
`https://digitalcourage.video/static/webseed/5a039eb2-9fbc-441d-a772-59ae3ee65c15-1080.mp4` has 6758472669 Bytes

So let's go REALLY big. EDIT: No direct (deep) link to a web-seed.

Quix0r committed:
Size "mediumint" wasn't enough:
https://digitalcourage.video/static/webseed/5a039eb2-9fbc-441d-a772-59ae3ee65c15-1080.mp4 has 6758472669 Bytes

So let's go REALLY big.

Quix0r commented:
Oh, why is `doc/database/db_post-media.md` failing check?

MrPetovan commented:
The best way to solve this problem is to pipe the result of `bin/console dbstructure dump` into `database.sql` and commit the result like this:
```
> bin/console dbstructure dump > database.sql
> git add database.sql
> git commit
```
Then you'll be sure the check succeeds.

Quix0r committed:
Fixed documentation/SQL dump with ./bin/console dbstructure dump

Quix0r commented:
`Failed asserting that two DOM documents are equal.`
Is now the failing one.

annando commented on static/dbstructure.config.php:
```diff
@@ -1257,7 +1257,7 @@
 			"height" => ["type" => "smallint unsigned", "comment" => "Height of the media"],
 			"width" => ["type" => "smallint unsigned", "comment" => "Width of the media"],
-			"size" => ["type" => "mediumint unsigned", "comment" => "Media size"],
+			"size" => ["type" => "bigint unsigned", "comment" => "Media size"],
```
We don't need 18 trillions. `int` is more than enough.

Quix0r replied:
No, `int` is clearly not enough for e.g. video files in Full-HD or 4k which are WAY larger than what `int` can hold.

MrPetovan replied:
`int unsigned` is 4 GB maximum, which is too small for video file byte sizes.

annando replied:
`int` is enough, (it is bigger than `mediumint`) have a look here: https://dev.mysql.com/doc/refman/8.0/en/integer-types.html

MrPetovan replied:
I made a mistake in my previous post, here's the maximum for each size and their corresponding size in bytes:
- `mediumint unsigned`: 16777215 so 16 MB max
- `int unsigned`: 4294967295 so 4GB max (not unusual for video files)
- `bigint unsigned`: 2^64-1 so ~8 zetta bytes which is an acceptable maximum for video file sizes.
 
`bigint` is the correct way forward to avoid these field size errors.

MrPetovan commented:
> There was 1 failure:
>
> 1\) Friendica\Test\src\Module\Api\Friendica\NotificationTest::testWithXmlResult
> Failed asserting that two DOM documents are equal.
> --- Expected
> +++ Actual
> @@ @@
> 
> \-
> \+
> 
> /woodpecker/src/github.com/friendica/friendica/tests/src/Module/Api/Friendica/NotificationTest.php:72

Quix0r commented:
I don't know what went wrong here and how to fix it.

MrPetovan commented:
Fair enough, it is a cryptic message. Let me see what I can do.

MrPetovan commented:
I checked out your branch, ran the faulty test and it turned out OK, not sure what's wrong.

MrPetovan reviewed (changes requested):
Please increment the database version number, both in `database.sql` and `statis/dbstructure.config.php`.

Quix0r commented:
Huh? Now all are passed? Maybe your clone did it? LOL

MrPetovan commented on database.sql:
```diff
@@ -1,6 +1,6 @@
 -- ------------------------------------------
 -- Friendica 2022.09-dev (Giant Rhubarb)
--- DB_UPDATE_VERSION 1469
+-- DB_UPDATE_VERSION 1470
```
Since I just merged your previous PR, you need to increment these again.

Quix0r replied:
Do you want me to rebase this branch to `develop`?

Quix0r committed:
Incremented again

Quix0r commented:
I rebased the changes as I got merge conflicts here while I didn't had any merge conflicts locally.

Okay, so for next round I know that I have to increment the 2 version numbers. PS: Doesn't `./bin/console dbstructure dumpsql >database.sql` update this?

annando commented on doc/database/db_post-media.md:
```diff
@@ -6,27 +6,27 @@ Attached media
+| --------------- | --------------------------------------------------------- | ----------------- | ---- | --- | ------- | -------------- |
+| id              | sequential ID                                             | int unsigned      | NO   | PRI | NULL    | auto_increment |
+| uri-id          | Id of the item-uri table entry that contains the item uri | int unsigned      | NO   |     | NULL    |                |
+| url             | Media URL                                                 | text              | NO   |     | NULL    |                |
```
The `url` field mustn't be a `text` since we use it in an unique index-

Quix0r replied:
Then we will have again the same error about being to short for super-long URLs like `magnet:` URLs can be. That's why I did these both changes, to cover long URLs and big video files. If I change both back to original vales we have the same two DB errors back.

annando reviewed (changes requested):
Please change the `text` back to the `varbinary` field that it had been before. We mustn't use `text` here since we use the field in an `unique` index.

annando commented:
Your previous PR changed the `int` to `mediumint`, but `mediumint` is small than `int`, see here: https://dev.mysql.com/doc/refman/8.0/en/integer-types.html

Quix0r commented:
@annando Okay, still `int` allows a maximum file size 4294967295 of Bytes which is nearly 40 GB (39.3 GB) which is for Full-HD videos to small to cover it. `bigint` allows them to be tracked in `post-media` table. Yes, maybe I didn't notice it that `mediumint` is smaller, still `int` isn't enough for some Full-HD videos, never to mention 4k/8k videos.

Quix0r committed:
Changes:
 - `url` to `varbinary(1024)`
 - `preview` to `varbinary(512)`

as the previous once (half length) were not enough in real-life scenarios

Quix0r committed:
Changes:
- added type-hints
- changed some double-quotes to single

MrPetovan commented:
I made a mistake in my previous post, here's the maximum for each size and their corresponding size in bytes:
- `mediumint unsigned`: 16777215 so 16 MB max
- `int unsigned`: 4294967295 so 4GB max (not unusual for video files)
- `bigint unsigned`: 2^64-1 so ~8 zetta bytes which is an acceptable maximum for video file sizes.
 
`bigint` is the correct way forward to avoid these field size errors.

MrPetovan commented on database.sql:
```diff
@@ -1,6 +1,6 @@
 -- ------------------------------------------
 -- Friendica 2022.09-dev (Giant Rhubarb)
--- DB_UPDATE_VERSION 1470
+-- DB_UPDATE_VERSION 1472
```
Why did you skip 1471?

Quix0r replied:
I had it internally and it was released on my server.
